### PR TITLE
Chitter Challenge

### PR DESCRIPTION
**Pull request by phump81**

### CODE REVIEW

File: spec/message_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require 'message'
+
+describe Message do
+
+  describe '.all' do
+    it 'shows a list of messages' do
+      connection = PG.connect(dbname: 'chitter_test')
+      message = Message.create(message: "The President is an idiot")
+      Message.create(message: "The PM is a bellend")
```
Comment (harrylb14) [inline spec/message_spec.rb:9]: Rate this, I would use this site

File: spec/setup_test_database.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require 'pg'
+
+def setup_test_database
+  connection = PG.connect(dbname: 'chitter_test')
```
Comment (harrylb14) [inline spec/setup_test_database.rb:4]: this could utilise your DatabaseConnection class rather than manually connecting to PG again

File: lib/user.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+class User
+
+  attr_reader :user_id, :name, :username, :email, :password
+
+  def initialize(user_id:, name:, username:, email:, password:)
+    @user_id = user_id
+    @name = name
+    @username = username
+    @email = email
+    @password = password
```
Comment (harrylb14) [inline lib/user.rb:12]: perhaps don't want password so easily accessible!

File: views/sign_up.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<form action="/sign_up" method="post">
+  <input type="text" name="name" placeholder="Name" />
+  <br>
```
Comment (harrylb14) [inline views/sign_up.erb:3]: use of <br> definitely makes it look much nicer!

File: spec/features/posts_new_message_spec.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+feature 'posting new messages' do
```
Comment (harrylb14) [inline spec/features/posts_new_message_spec.rb:1]: perhaps could do with a few more feature tests throughout? perhaps may help for edge cases etc

File: spec/setup_test_database.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+  connection.exec("TRUNCATE messages;")
```
Comment (harrylb14) [inline spec/setup_test_database.rb:5]: useful tip i found last week, if you do "TRUNCATE messages, RESTART IDENTITY;" it resets the id counter as well. as currently everytime you reset it, while it wipes the table it still keeps counting from where you left off, so after a few rspecs youll be on silly numbers in the ID column. restart identity resets it to 1 each time. probably not essential, but quite nice I thought!

File: app.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
+require 'sinatra/base'
```
Comment (harrylb14) [inline app.rb:1]: v skinny controller, we love to see it
